util/chplenv/chpl_compiler.py

```diff
@@ -198,12 +198,6 @@ def get(flag='host'):
                 compiler_val = 'clang'
             else:
                 compiler_val = 'gnu'
-        elif locale_model_val == 'gpu':
```

Comment:
I assume this check only gets invoked if the user didn't explicitly specify the target compiler (previously it looks like it would default to gnu except under certain circumstances). You've modified it to default to clang (if locale model is gpu) and error out if we don't find clang.

Do you know if we have an error triggered if the user explicitly sets the compiler to gnu and tries to compile with the gpu locale model?

Reply:
There is no error if the user explicitly sets CHPL_TARGET_COMPILER to something other than llvm/clang. I opened an issue https://github.com/Cray/chapel-private/issues/3033 suggesting an error or warning.